teps/0091-trusted-resources.md

```diff
@@ -270,18 +256,38 @@ metadata:
  namespace: tekton-pipelines
 spec:
    resources:
+   # resources is used to filter out remote resources by their sources URL. e.g. if the resources from https://github.com/tektoncd/catalog.git, then the it should be set to https://github.com/tektoncd/catalog.git.
     - https://github.com/tektoncd/catalog.git
     - gcr.io/tekton-releases/catalog/upstream/*
+  # authoritiyList defines a list of keys
    authorities:
-    - name: cosign
-      key: /etc/signing-secrets/cosign.pub
+    - name: key_in_secret
+      key:
+        # secretRef refers to a secret in the cluster, this secret should contain public keys data
+        secretRef:
+          name:  tekton-pipelines
+          namespace: verification-secrets
+        # hashAlgorithm is the algorithm for the public key, by default is "sha256"
+        hashAlgorithm: sha256
+    - name: key_in_data
+        key:
+          # data is string format of the public key data. This can be configured by users, or if using the secretRef the data is populated and will be used next time before looking up the secret. This can reduce the api calls.
+          data: "STRING_ENCODED_PUBLIC_KEY"
     - name: kms
       key: gcpkms://projects/<project>/locations/<location>/keyRings/<keyring>/cryptoKeys/<key>
-    - name: keyless
-      keyless:
-        url: "https://fulcio.sigstore.dev"
 ```
 
+`namespace` should be the same of corresponding resources' namespace.
+
+`pattern` is used to filter out remote resources by their sources URL. e.g. git resources pattern can be set to https://github.com/tektoncd/catalog.git. The `pattern` should follow regex schema, we use go regex library's [`Match`](https://pkg.go.dev/regexp#Match) to match the pattern from VerificationPolicy to the `ConfigSource` URL resolved by remote resolution. Note that `.*` will match all resources.
```

Comment:
I don't see `pattern` in the example above.
Is filtering possible only for remote resources? What about resources in the cluster?

Reply:
resources in the cluster are also supported by using clusterresolver. 
https://github.com/tektoncd/pipeline/blob/main/docs/cluster-resolver.md#resolutionrequest-status
We can use the configsource to filter

Reply:
cluster resources verification are also supported if using the local resolver, the only missing part is that is doesn't has source information, so we cannot use a pattern to filter it. In that case users can create one policy to match all cluster resources.